contracts/RandomConsumer.cdc

```diff
@@ -0,0 +1,295 @@
+import "Burner"
+
+import "RandomBeaconHistory"
+import "Xorshift128plus"
+
+/// This contract is intended to make it easy to consume randomness securely from the Flow protocol's random beacon. It provides
+/// a simple construct to commit to a request, and reveal the randomness in a secure manner as well as helper functions to
+/// generate random numbers in a range without bias.
+///
+/// See an example implementation in the repository: https://github.com/onflow/random-coin-toss
+///
+access(all) contract RandomConsumer {
+
+    /* --- PATHS --- */
+    //
+    /// Canonical path for Consumer storage
+    access(all) let ConsumerStoragePath: StoragePath
+
+    /* --- EVENTS --- */
+    //
+    access(all) event RandomnessRequested(requestUUID: UInt64, block: UInt64)
+    access(all) event RandomnessSourced(requestUUID: UInt64, block: UInt64, randomSource: [UInt8])
+    access(all) event RandomnessFulfilled(requestUUID: UInt64, randomResult: UInt64)
+
+    ///////////////////
+    // PUBLIC FUNCTIONS
+    ///////////////////
+
+    /// Retrieves a revertible random number in the range [min, max]. By leveraging the Cadence's revertibleRandom
+    /// method, this function ensures that the random number is generated within range without risk of bias.
+    ///
+    /// @param min: The minimum value of the range
+    /// @param max: The maximum value of the range
+    ///
+    /// @return A random number in the range [min, max]
+    ///
+    access(all) fun getRevertibleRandomInRange(min: UInt64, max: UInt64): UInt64 {
```

Comment:
This function isn't used by the project example right? just making sure I didn't miss anything :)

Reply:
Correct, added for parity with the solidity base contract and for demonstration but not used in the context of this repo.